### PR TITLE
1246 refactor site settings

### PR DESCRIPTION
**Pull request by dudethatbe**
moved the settings in the chat to be in the main menu. i got stuck for awhile because of the sp and pc nav bars. not really a fan of the repeated code, but that's what works with these two components. we'll probably end up refactoring the menus anyways so 🤷‍♂️

### desktop
pc nav bar with config button

### mobile
sp nav bar with config button

### CODE REVIEW

**Comment by dudethatbe** [comment]
I copied the Locale, Theme, and Weather dropdowns into the menu and only the theme menu is working so far
![updated settings dialog]https://github.com/datafruits/datafruits/assets/2829501/de85e605-3cd1-48d2-847a-1725f6db9f3c)

File: app/components/website-settings.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import Component from "@glimmer/component";
+import { inject as service } from "@ember/service";
+import { action } from "@ember/object";
+
+import ChatService from "datafruits13/services/chat";
```
Comment (mcfiredrill) [inline app/components/website-settings.ts:5]:
```suggestion
import type ChatService from "datafruits13/services/chat";
```

File: app/templates/components/website-settings.hbs [hunk 2]
```diff
@@ -0,0 +1 @@
+{{yield}}
```
Comment (mcfiredrill) [inline app/templates/components/website-settings.hbs:1]: delete this file

File: app/components/website-settings.ts (hunk 1 above)
Comment (dudethatbe) [reply app/components/website-settings.ts]: doops thanks for pointing that out

**Comment by dudethatbe** [comment]
some comments from discord
[] change buttons to checkboxes so they align vertically with the other settings
[] move ⚙ to right corner

**Comment by dudethatbe** [comment]
updated settings form where the user and gif options are checkboxes now
here's what I got so far. need to test the viz toggle, but that should be nearly done. think I need to make some more substantive changes for the gear icon to appear on the right because that's the home template instead of the navbars. which is kinda nice because that means less repeated code from the PC and sp navbar components 😅😓

**Comment by dudethatbe** [comment]
kinda clunky but the settings show up on the right if you're logged in 
logged in settings button

or if you're not logged in
settings button next to 3D button

or if on mobile
settings button at the end of the menu on mobile

File: app/components/chat/enable-gifs-button.hbs [hunk 3]
```diff
@@ -1,7 +1,18 @@
...
-  {{on "click" @toggleGifs}}>
-  {{if @enabled this.imgOn this.imgOff}}
-</button>
\ No newline at end of file
+>
+  <label
+    id="gif-enabler-label"
+    for="gif-enabler"
+  >
+    Show GIFs
```
Comment (mcfiredrill) [inline app/components/chat/enable-gifs-button.hbs:9]: Guess it's technically show images? We might have a translation key for this already

Comment (dudethatbe) [reply app/components/chat/enable-gifs-button.hbs:9]: Went thru the translation file and there's only labels for the title that changes ("Showing Images" and "Not Showing Images"). But yeah the text shouldn't be hardcoded in there anyways

File: app/components/chat/tv-mode-button.hbs [hunk 4]
```diff
@@ -1,8 +1,18 @@
...
-  </button>
-{{/if}}
+    Show Video
+  </label>
+  <input 
+    type="checkbox"
+    name="tv-mode-input"
+    id="tv-mode-input"
+    {{on "change" @toggleVideo}}
+    checked=@enabled
```
Comment (mcfiredrill) [inline app/components/chat/tv-mode-button.hbs:16]: 2024-01-21-211929_885x349_scrot

When I have both settings windows open , looks like the checkbox on one doesn't update....even tho we are using the service so it seems like it should , weird.

Comment (mcfiredrill) [inline app/components/chat/tv-mode-button.hbs:16]:
```suggestion
    checked={{@enabled}}
```

File: app/components/chat/enable-gifs-button.hbs [hunk 5]
```diff
@@ -1,7 +1,18 @@
...
+  </label>
+  <input 
+    id="gif-enabler"
+    name="gif-enabler"
+    type="checkbox"
+    checked=@enabled
```
Comment (mcfiredrill) [inline app/components/chat/enable-gifs-button.hbs:15]:
```suggestion
    checked={{@enabled}}
```

File: app/components/chat/tv-mode-button.hbs (hunk 4 above)
Comment (mcfiredrill) [reply app/components/chat/tv-mode-button.hbs:16]: My suggestion below might fix it

Comment (dudethatbe) [reply app/components/chat/tv-mode-button.hbs]: nice it works now 🥳

Comment (dudethatbe) [reply app/components/chat/tv-mode-button.hbs]: Thanks those fixed it. Hard to remember all these ember things 😅

**Review by mcfiredrill — APPROVED** [review]

**Pull request merged**